lib/scorecard.rb

```
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require 'frame'
+
+class ScoreCard
+  LAST_FRAME = 10
+  TEN_PINS = 10
+  ZERO_SCORE = 0
+  SPARE = 1
+  STRIKE = 2
+
+  def initialize(frame_class = Frame, last_frame_class = LastFrame)
+    @frame_class = frame_class
+    @last_frame_class = last_frame_class
+    @frames = {}
+  end
+
+  def current_frame_number
+    @frames.count
+  end
+
+  def log_roll(pins_downed)
+    return :game_complete if game_complete?
```

Trying to enter a roll to a completed game should probably raise an exception.